### ISSUE TITLE
Is template feature not working in expected. Is it bug?

### ISSUE THREAD

**Issue by netpyoung** [vaab/gitchangelog#48]
i installed gitchangelog by `sudo pip install gitchagelog`

and i try test `template` feature.
but, i can't get what i expected.

here is some example

```
[~/temp/gitchangelog]$ uname -a                                      *[master]
Darwin pyoung.local 15.5.0 Darwin Kernel Version 15.5.0: Tue Apr 19 18:36:36 PDT 2016; root:xnu-3248.50.21~8/RELEASE_X86_64 x86_64

[~/temp/gitchangelog]$ diff .gitchangelog.rc gitchangelog.rc.reference
181c181
< #output_engine = rest_py

---
> output_engine = rest_py
183c183
< output_engine = mustache("markdown")

---
> #output_engine = mustache("markdown")


[~/temp/gitchangelog]$ gitchangelog                                  *[master]
Error: Invalid mustache template name 'markdown'.
No available mustache templates found in '/usr/local/lib/python2.7/site-packages/templates/mustache'.


[~/temp/gitchangelog]$ ./gitchangelog                                *[master]
# Changelog

## 2.3.0 (2015-09-25)

### Fix
.
.
.
```

i was clone gitchangelog to ~/temp/gitchangelog
but when i try to generate by `gitchangelog` not `./gitchangelog`, it's failed.

i supposed that caused by `__file__` path based.

https://github.com/vaab/gitchangelog/blob/master/gitchangelog.py#L663

``` python
template_dir = os.path.join(
        os.path.dirname(os.path.realpath(__file__)),
        "templates", label)
```

---
- Is this working in expected? or Bug?
- In additional, Could you add feature that specifiy template_dir from `.gitchangelog.rc` by optional.

---

sorry, i found pull request about that. #46

**Comment by vaab** [comment]
There's a definite packaging bug here, again. sigh ....

If by chance you have 2min to check if this is still the case, I would be glad to know your version of pip and setuptools. I'll dig into this anyway hopefull before next week.

**Comment by netpyoung** [comment]
thanks to replay.

I formated my macbook. but it also reproduced.

~~~
[~/temp/gitchangelog]$ git st                                        *[master]
On branch master
Your branch is up-to-date with 'origin/master'.
Untracked files:
  (use "git add <file>..." to include in what will be committed)

       	.gitchangelog.rc
       	.python-version

nothing added to commit but untracked files present (use "git add" to track)

[~/temp/gitchangelog]$ diff .gitchangelog.rc gitchangelog.rc.reference
181c181
< #output_engine = rest_py
---
> output_engine = rest_py
183c183
< output_engine = mustache("markdown")
---
> #output_engine = mustache("markdown")

[~/temp/gitchangelog]$ git rev-parse HEAD                            *[master]
e9759c2ed7baf415d64632541d4ae0f0fde696b5

[~/temp/gitchangelog]$ uname -a                                      *[master]
Darwin pyoungui-mbp 15.6.0 Darwin Kernel Version 15.6.0: Mon Aug 29 20:21:34 PDT 2016; root:xnu-3248.60.11~1/RELEASE_X86_64 x86_64

[~/temp/gitchangelog]$ python --version                              *[master]
Python 2.7.6

[~/temp/gitchangelog]$ cat .python-version                           *[master]
2.7.6

[~/temp/gitchangelog]$ pip --version                                 *[master]
pip 8.1.2 from /Users/pyoung/.pyenv/versions/2.7.6/lib/python2.7/site-packages (python 2.7)

[~/temp/gitchangelog]$ gitchangelog                                  *[master]
Required 'pystache' python module not found.


[~/temp/gitchangelog]$ pip install pystache                          *[master]
Collecting pystache
/Users/pyoung/.pyenv/versions/2.7.6/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:318: SNIMissingWarning: An HTTPS request has been made, but the SNI (Subject Name Indication) extension to TLS is not available on this platform. This may cause the server to present an incorrect TLS certificate, which can cause validation failures. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#snimissingwarning.
  SNIMissingWarning
/Users/pyoung/.pyenv/versions/2.7.6/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:122: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
  Downloading pystache-0.5.4.tar.gz (75kB)
    100% |████████████████████████████████| 81kB 3.3MB/s
Building wheels for collected packages: pystache
  Running setup.py bdist_wheel for pystache ... done
  Stored in directory: /Users/pyoung/Library/Caches/pip/wheels/cb/7f/2c/c0cbdfc2fc99bebbe38218110b53ba8282c4ad9f7cfaa2bc21
Successfully built pystache
Installing collected packages: pystache
Successfully installed pystache-0.5.4


[~/temp/gitchangelog]$ gitchangelog                                  *[master]
Error: Invalid mustache template name 'markdown'.
No available mustache templates found in '/Users/pyoung/.pyenv/versions/2.7.6/lib/python2.7/site-packages/templates/mustache'.


[~/temp/gitchangelog]$ ./gitchangelog                                *[master]
/usr/local/Cellar/pyenv/1.0.2_1/libexec/pyenv: line 43: cd: gitchangelog.py: Not a directory
# Changelog

## 2.3.0 (2015-09-25)

### Fix

* Nasty hidden bug that would break python3 (fixes #27) [Valentin Lab]

  Actually thi
~~~

**Comment by vaab** [comment]
I'm happy to announce that version 3.0.0 that was just released should not have anymore these issues and moved to a saner packaging scheme. Thanks for your report !
